tests/Java.Interop.Tools.JavaSource-Tests/SourceJavadocToXmldocParserTests.cs

```diff
@@ -144,16 +144,16 @@ @apiSince 1
 </member>",
 			},
 			new ParseResult {
-				Javadoc = "Something {@link #method}: description, \"<code>declaration</code>\" or \"<code>another declaration</code>\".\n\n@apiSince 1\n",
```

Comment:
As per the PR message:

> [we] no longer attempting to parse `<code>` elements with Irony

but the Javadoc we parse may (will) still contain `<code/>` elements.  Thus the question: *what happens*?

Changing this line means I (the reviewer) cannot easily determine what happens when we "no longer attempt to parse".  *I assume* that given the input:

```
Something {@link #method}: description, "<code>declaration</code>" or "<code>another declaration</code>".

@apiSince 1
```

this PR will now generate:

```xml
<member>
  <summary>Something <c>#method</c>: description, "&lt;code&gt;declaration&lt;/code&gt;" or "&lt;code&gt;another declaration&lt;/code&gt;".</summary>
</member>
```

which would be useful to know!

Additionally, the PR message states:

> The latest API docs update contained a couple dozen parsing issues due
to broken `<code></code>` elements, reserved inline characters in
`<code>` elements, and other issues.

We should add unit tests which reproduce these issues.  What constitutes a "broken `<code></code>` element"?  *What kind* of "reserved inline characters" were observed?  What other issues?

Reply:
This assumption is mostly correct, however the HTML fix up attempts added to generator in this PR would ultimately produce this doc:

```xml
<member>
  <summary>Something <c>#method</c>: description, "<c>declaration</c>" or "<c>another declaration</c>".</summary>
</member>
```

The couple dozen corner cases of "broken" code elements were caused by the following types of issues:

* `<code>null</null>` - wrong closing tag
* `<code>android:label="@string/resolve_title"</code>` - `@` tag rule seemingly taking precedence
* `<code>Activity.RESULT_OK<code>` - trailing open tag rather than closing tag
* `<code><pre><p>content</code></pre></p>` - additional inline html tags

Reply:
I've restored this portion of the test, and extended it to contain the very odd `<code><pre><p>content</code></pre></p>` corner case. `SourceJavadocToXmldocGrammar.HtmlBnfTermsTests.cs` has been updated to include all of the other failure cases.